lib/linguist/generated.rb

```diff
@@ -244,7 +244,11 @@ def generated_parser?
     #
     # Returns true or false.
     def generated_postscript?
-      return false unless ['.ps', '.eps'].include? extname
+      return false unless ['.ps', '.eps', '.pfa'].include? extname
+
+      # Type 1 and Type 42 fonts converted to PostScript are stored as hex-encoded byte streams; these
+      # streams are always preceded the `eexec` operator (if Type 1), or the `/sfnts` key (if Type 42).
+      return true if data =~ /(\n|\r\n|\r)\s*(?:currentfile eexec\s+|\/sfnts\s+\[\1<)\h{8,}\1/
```

Comment:
Since you like explanations, I'll also add that at least 8 hexadecimal characters are matched, because the Adobe Type 1 specification mandates that encoded `eexec` sections are prefixed with four random, non-ASCII bytes for encryption purposes. So `4 * 2 = 8`. I imagine this was initially done to keep the format secret (recalling this was once proprietary software back before they open-sourced it).